cmd/lncli/commands.go

```diff
@@ -538,6 +569,190 @@ func showRoutingTable(ctx *cli.Context) error {
+	} else {
+		printRTAsJSON(r)
+	}
 	return nil
 }
+
+func writeToTempFile(r *rt.RoutingTable, file *os.File, self []byte) {
+	slc := []graph.ID{graph.NewID(string(self))}
+	viz := visualizer.New(r.G, slc, nil, nil)
+	viz.ApplyToNode = func(s string) string { return hex.EncodeToString([]byte(s)) }
+	viz.ApplyToEdge = func(info interface{}) string { 
+		if info, ok := info.(*rt.ChannelInfo); ok {
+			return fmt.Sprintf(`"%v"`, info.Capacity())
+		}
+		return "nil"
+	}
+	// need to call method if plan to use shortcut, autocomplete, etc
+	viz.BuildPrefixTree()
+	viz.EnableShortcut(true)
+	dot := viz.Draw()
+	file.Write([]byte(dot))
+	file.Sync()
+}
+
+func writeToImageFile(TempFile, ImageFile *os.File) {
+	visualizer.Run("neato", TempFile, ImageFile)
+	TempFile.Close()
+	os.Remove(TempFile.Name())
+	ImageFile.Close()
+}
+
+// get around a bug in the standard library, add suffix param
+func TempFileWithSuffix(dir, prefix, suffix string) (f *os.File, err error) {
+	f, err = ioutil.TempFile(dir, prefix)
+	if err != nil {
+		return nil, err
+	}
+	defer os.Remove(f.Name())
+	f, err = os.Create(f.Name()+suffix)
+	return
+}
+
+// Prints routing table in human readable table format
+func printRTAsTable(r *rt.RoutingTable, humanForm bool) {
+	// Minimum length of data part to which name can be shortened
+	var minLen int
+	var tmpl string
+	var lightningIdTree, edgeIdTree prefix_tree.PrefixTree
+	if humanForm {
+		tmpl = "%-10v %-10v %-10v %-10v %-10v\n"
+		minLen = 6
+	} else {
+		tmpl = "%-64v %-64v %-66v %-10v %-10v\n"
+		minLen = 100
+	}
+	fmt.Printf(tmpl, "ID1", "ID2", "EdgeID", "Capacity", "Weight")
+	channels := r.AllChannels()
+	if humanForm {
+		// Generate prefix tree for shortcuts
+		lightningIdTree = prefix_tree.NewPrefixTree()
+		for _, node := range r.Nodes() {
+			lightningIdTree.Add(hex.EncodeToString([]byte(node.String())))
+		}
+		edgeIdTree = prefix_tree.NewPrefixTree()
+		for _, channel := range channels {
+			edgeIdTree.Add(channel.EdgeID.String())
+		}
+	}
+	for _, channel := range channels {
+		var source, target, edgeId string
+		sourceHex := hex.EncodeToString([]byte(channel.Id1.String()))
+		targetHex := hex.EncodeToString([]byte(channel.Id2.String()))
+		edgeIdRaw := channel.EdgeID.String()
+		if humanForm {
+			source = getShortcut(lightningIdTree, sourceHex, minLen)
+			target = getShortcut(lightningIdTree, targetHex, minLen)
+			edgeId = getShortcut(edgeIdTree, edgeIdRaw, minLen)
+		} else {
+			source = sourceHex
+			target = targetHex
+			edgeId = edgeIdRaw
+		}
+		fmt.Printf(tmpl, source, target, edgeId, channel.Info.Cpt, channel.Info.Wgt)
+	}
+}
+
+func getShortcut(tree prefix_tree.PrefixTree, s string, minLen int) string {
+	s1, err := tree.Shortcut(s)
+	if err != nil || s == s1 {
+		return s
+	}
+	if len(s1) < minLen && minLen < len(s) {
+		s1 = s[:minLen]
+	}
+	shortcut := fmt.Sprintf("%v...", s1)
+	if len(shortcut) >= len(s) {
+		shortcut = s
+	}
+	return shortcut
+}
+
+func printRTAsJSON(r *rt.RoutingTable) {
```

Comment:
It decouples representation used for transmission of routing table from representation used by user. If for example some additional info should be added to this table (like distance from our node to channel) than it can be calculated directly in lncli without modifying rpc interface. Sure, printRespJson can be used directly but it will increase coupling.